### PR TITLE
Set OS/2 unicode ranges

### PR DESCRIPTION
**Pull request by rsheeter**
OS/2 flags are more fiddly than I gave them credit for

### CODE REVIEW

File: fontbe/src/os2.rs [hunk 1]
```diff
@@ -10,6 +12,187 @@ use crate::{
     orchestration::{BeWork, Context},
 };
 
+/// Used to build a bitfield.
+///
+/// Intended use: use the outer range to determine if a given bit might apply. If it might,
+/// check the inner ranges.
```
Comment (anthrotype) [inline fontbe/src/os2.rs:18]: the comment is no longer in sync with the implementation below I think, I don't see outer/inner ranges anymore, just a single array of 3-tuples

**Review by anthrotype — APPROVED** [review]
LGTM!

File: fontbe/src/os2.rs [hunk 2]
```diff
@@ -10,6 +12,187 @@ use crate::{
...
+///
+/// Generated by resources/scripts/os2_unicode_ranges.py.
+///
+/// <https://learn.microsoft.com/en-us/typography/opentype/spec/os2#ur>
+/// <https://github.com/fonttools/fonttools/blob/47813b217c1f8bc343c094776020b4f32fc024b0/Lib/fontTools/ttLib/tables/O_S_2f_2.py#L351-L557>
+const UNICODE_RANGES: &[(usize, usize, usize)] = &[
```
Comment (cmyr) [inline fontbe/src/os2.rs:24]: is `usize` the best type to represent a codepoint? `char` might be semantically more accurate, and `u32` would be a more obvious choice if we want to use an integer type.

Also I wonder if it makes sense to represent the ranges as an actual range? maybe not, but just a thought.

File: fontbe/src/os2.rs [hunk 3]
```diff
@@ -80,6 +263,51 @@ fn apply_metrics(os2: &mut Os2, metrics: &GlobalMetricsInstance) {
...
+    let maybe_idx = UNICODE_RANGES
+        .binary_search_by(|(from, to, _)| match codepoint {
+            _ if codepoint < *from => Ordering::Greater,
+            _ if codepoint > *to => Ordering::Less,
+            _ => Ordering::Equal,
+        })
+        .ok();
+    if let Some(idx) = maybe_idx {
+        add_to.insert(UNICODE_RANGES[idx].2);
+    }
```
Comment (cmyr) [inline fontbe/src/os2.rs:277]: code golf:

```suggestion
    // Is there any range this codepoint _might_ fit?
    if let Ok(idx) = UNICODE_RANGES
        .binary_search_by(|(from, to, _)| match (from.cmp(&codepoint), to.cmp(&codepoint)) {
        (Ordering::Greater, _) => Ordering::Greater,
        (_, Ordering::Less) => Ordering::Less,
        _ => Ordering::Equal,
        }) {
            add_to.insert(UNICODE_RANGES[idx].2);
        }      
    }
```

**Review by cmyr — APPROVED** [review]
Some notes inline that you are very welcome to ignore, overall looks reasonable!

File: fontbe/src/os2.rs (hunk 2 above)
Comment (rsheeter) [reply fontbe/src/os2.rs:24]: Fixed

File: fontbe/src/os2.rs (hunk 3 above)
Comment (rsheeter) [reply fontbe/src/os2.rs:277]: I tried that but I find it harder to read than doing it in separate steps and afaik there is no downside?

Comment (rsheeter) [reply fontbe/src/os2.rs:277]: ...the comment is out of date!

Comment (rsheeter) [reply fontbe/src/os2.rs]: I tried it again adn I still think it's harder to read so I'm inclined to leave this as-is.

**Pull request merged**